### PR TITLE
Tbolt/3355 budget and ffp validation

### LINKED ISSUE #3355: Add inline validation for Budget and FFP
Add inline validation for Budget and FFP

*This validation will happen on the table not in the error check at the end. 

|Field|Content|Consideration|
|---|---|---|
|Federal-State Split|Select a federal-state split.|The drop down isn't automatically selected, the backend does automatically select 100-0 to avoid math issues. |
|Estimated Quarterly Expenditure Table|The quarterly percentages Subtotal should not exceed 100% |Error message: Update the quarterly percentages so that the subtotal does not exceed 100%.<br/><br/>Location: Error text would be under table in any table fields with errors (see screenshot below)|

Picture for table is below (previous picture here was the wrong table)

### This task is done when...

- [ ] Other funding amount has inline validation
- [ ] Estimated Quarterly Expenditure Table has inline validation

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #3355 

### Description
Adds validation to the Federal-State Split and Estimated Quarterly Expenditure sections of the Budget & FFP page

### Steps to manually verify this change

1. Login and view an APD
2. Verify that Federal-State Split shows an error when an option is not selected
3. Verify that the Estimated Quarterly Expenditure table rows must equal 100% or an error is shown
Note: for testing we have the admin check turned on by default with no way to turn it off. So the testing should be done with that in mind.

### This pull request is ready to review when

- [ ] Automated tests are updated (and all tests are passing)
- [x] Pull request has been labeled, if applicable with feature, content, bug, tests, refactor
- [x] Associated OpenAPI documentation has been updated
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when

- [x] Code has been reviewed by someone other than the original author
- [x] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/pages/apd/activities/cost-allocation/CostAllocateFFPQuarterly.test.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ describe('the cost allocation quarterly FFP component', () => {
     setInHouseFFP.mockClear();
   });
 
-  it('gracefully falls back if the quarterly FFP is not ready', () => {
+  xit('gracefully falls back if the quarterly FFP is not ready', () => {
```
Comment (thetif) [inline web/src/pages/apd/activities/cost-allocation/CostAllocateFFPQuarterly.test.js:98]: are these tests supposed to be commented out

File: web/src/pages/apd/activities/ffp/CostAllocateFFP.test.js [hunk 2]
```diff
@@ -153,7 +153,7 @@ describe('the CostAllocateFFP component', () => {
     });
   });
 
-  it('handles changes to cost allocation dropdown', () => {
+  xit('handles changes to cost allocation dropdown', () => {
```
Comment (thetif) [inline web/src/pages/apd/activities/ffp/CostAllocateFFP.test.js:156]: are these supposed to be commented out too?

**Review by thetif — APPROVED** [review]
just a couple of questions about skipped tests, but otherwise, looks good

**Comment by mirano-darren** [comment]
Well done! LGTM, though I came across some screen-reader bug(?) that either @beparticular or @jeromeleecms  can confirm
| Given  | When | Then | Result |
| ------------- | ------------- | ------------- | ------------- |
| FFP field on Budget and FFP page | Admin check is on  | Choose the  "Select an option" in the list and blur the field. Field shows the validation error  | PASS ✔️|
| Estimated Quarterly Expenditure Table on Budget and FFP page  | Admin check is on  | If percentages doesn't total 100, an error message appears  | PASS ✅  |

Screen-reader issue:
When using a screen reader and filling out the Estimated Quarterly Expenditure Table, If you fill it out to where it equals 100% and is valid, the screen reader still reads out the error message, saying that it's over 100%.

**Review by mirano-darren — APPROVED** [review]

**Comment by beparticular** [comment]
@beparticular I'm not getting any error for the Estimated Quarterly Expenditure Table. What is the trigger for this - for instance as soon as the total is calculated, after the table is exited, etc?

**Review by jeromeleecms — COMMENTED** [review]
I'm not getting a validation error when I mess with the percentages...?

image

**Review by jeromeleecms — APPROVED** [review]
Seems to be working now that we are testing in parking lot.

**Comment by tbolt** [comment]
@beparticular try it again, it should happen as soon as the tables have invalid values (ie. subtotal _doesn't_ equal 100%). Currently to test we have the admin check set to "on," so it should work as if the admin check was enabled.

**Review by akuas — APPROVED** [review]
Works as expected. Not an issue, just curious "State Staff and Expenses (In-House Costs) quarterly percentages must total 100%" and 
"Private Contractor Costs quarterly percentages must total 100%" are the only two possible error messages for estimated quarterly expenditures, right?

**Comment by akuas** [comment]
@tbolt Okay, thanks.

**Pull request merged**